packages/graphql/src/mutations/_txHelper.js

```diff
@@ -140,6 +142,437 @@ async function fetchGasPrice() {
+  if (String(shouldUseProxy).startsWith('create')) {
+    resetProxyCache()
+  }
+
+  pubsub.publish('TRANSACTION_UPDATED', {
+    transactionUpdated: {
+      id: receipt.transactionHash,
+      status: 'receipt',
+      gasUsed: receipt.gasUsed,
+      mutation
+    }
+  })
+
+  if (contracts.automine) {
+    setTimeout(() => mineBlock(contracts.web3), contracts.automine)
+  }
+}
+
+/**
+ * Transaction confirmation handler. Used as an internal callback. Each
+ * "confirmation" is a mined block *after* the transaction was included in a
+ * block
+ *
+ * @param {object} args
+ * @param {string} args.shouldUseProxy - Return from useProxy function
+ * @param {number} args.confNumber - The amount of confirmations since a transaction
+ * @param {object} args.receipt - The receipt object from web3.js
+ * @param {string} args.mutation - The graphql mutation being executed
+ * @returns {undefined}
+ */
+async function handleConfirmation({
+  shouldUseProxy,
+  confNumber,
+  receipt,
+  mutation
+}) {
+  if (confNumber === 1) {
+    if (String(shouldUseProxy).indexOf('create') === 0) {
+      resetProxyCache()
+    }
+  }
+  if (confNumber > 0) {
+    pubsub.publish('TRANSACTION_UPDATED', {
+      transactionUpdated: {
+        id: receipt.transactionHash,
+        status: 'confirmed',
+        mutation,
+        confirmations: confNumber
+      }
+    })
+  }
+}
+
+/**
+ * Send a trnsaction using the Origin relayer
+ *
+ * @param {object} args
+ * @param {object} args.web3 - a web3.js instance
+ * @param {object} args.tx - a web3.js transaction object
+ * @param {string} args.proxy - the proxy address to use
+ * @param {string} args.sourceAccount - the originating Ethereum account
+ * @param {string} args.to - the destination Ethereum address
+ * @param {Array} args.hashCallbacks - Array of functions to use as callbacks for transaction hash
+ * @param {Array} args.receiptCallbacks - Array of functions to use as callbacks for transaction receipts
+ * @param {Array} args.confirmCallbacks - Array of functions to use as callbacks for transaction confirmations
+ * @returns {string} transaction hash
+ */
+async function sendViaRelayer({
+  web3,
+  tx,
+  proxy,
+  sourceAccount,
+  to,
+  hashCallbacks,
+  receiptCallbacks,
+  confirmCallbacks
+}) {
+  const resp = await relayer({
+    tx,
+    proxy,
+    from: sourceAccount,
+    to
+  })
+
+  if (!resp) {
+    throw new Error('No response from relayer!')
+  }
+
+  const txHash = resp.id
+
+  if (hashCallbacks && hashCallbacks.length) {
+    await handleCallbacks({
+      callbacks: hashCallbacks,
+      val: txHash
+    })
+  }
+
+  const hasCallbacks =
+    receiptCallbacks.length > 0 || confirmCallbacks.length > 0 ? true : false
+
+  /**
+   * Since we don't get the event handlers when the relayer processes a
+   * transaction, make sure that any provided callbacks are run by
+   * manually listening for new blocks.
+   */
+  let foundReceipt = false
+  if (hasCallbacks) {
```

Comment:
you can use early returns to decrease code indentation:
```
if (!hasCallbacks) {
  return  txhash
}
// remainder
```

Reply:
I agree when everything is in view, but somewhat disagree when there's a handful of returns in random spots within a function.  When debugging someone may be less likely to notice an early return.  This function is borderline, but I will keep this in mind in the future.